linkcheck/checker/urlbase.py

```diff
@@ -631,12 +631,13 @@ def get_raw_content(self):
     def get_content(self):
         if self.text is None:
             self.get_raw_content()
-            self.soup = htmlsoup.make_soup(self.data)
+            self.soup = htmlsoup.make_soup(self.data, self.encoding)
             # Sometimes soup.original_encoding is None!  Better mangled text
             # than an internal crash, eh?  ISO-8859-1 is a safe fallback in the
             # sense that any binary blob can be decoded, it'll never cause a
             # UnicodeDecodeError.
             self.encoding = self.soup.original_encoding or 'ISO-8859-1'
+            log.debug(LOG_CHECK, "Content encoding %s", self.encoding)
```

Comment:
Debug-wise it might be worth distinguishing between BeautifulSoup detecting ISO-8859-1 and our fallback kicking in.

Reply:
Added an extra log message for self.soup.original_encoding.